toc/rfc/rfc-draft-system-cnb.md

```diff
@@ -0,0 +1,26 @@
+# Meta
+[meta]: #meta
+- Name: System Cloud Native Buildpacks
+- Start Date: 30.04.2024
+- Author(s): @c0d1ngm0nk3y, @pbusko, @nicolasbender, @modulo11
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+- Predecessor: https://github.com/cloudfoundry/community/pull/796
+
+## Summary
+
+This RFC proposes extending the Cloud Foundry functionality to introduce and support system Cloud Native Buildpacks (CNBs). It will allow users to initiate the `cf push --lifecycle cnb` command without providing an explicit list of buildpacks, relying instead on the list of system buildpacks and the built-in auto-detection mechanism.
+
+## Problem
+
+Despite the recent integration of CNBs in the Cloud Foundry platform, the feature still has limitations that could hinder user experience and efficiency. Currently, only custom buildpacks are supported. Users have to manually provide a list of buildpacks (at least one) for their application to run with the `cnb` lifecycle type.
+
+## Proposal
```

Comment:
We should also state the default system buildpacks are the classical ones. I mean `cf push` will still iterate over the classical system buildpacks.

Reply:
Like https://github.com/cloudfoundry/community/pull/830/files/084872630c47b5b5a17ebddaa94d31e346c3ee5b#r1586535639 @beyhan?

Reply:
yes, thanks!